cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -704,6 +707,60 @@ public Void apply(AppMds mds) throws Exception {
     });
   }
 
+  @Override
+  public void deleteSchedules(final Id.Program program) {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        ApplicationSpecification existingAppSpec = getAppSpecOrFail(mds, program);
+        ApplicationSpecification newAppSpec =
+          removeSchedulesFromApp(existingAppSpec, new Predicate<ScheduleProgramInfo>() {
+            @Override
+            public boolean apply(@Nullable ScheduleProgramInfo scheduleProgramInfo) {
+              return scheduleProgramInfo != null && scheduleProgramInfo.getProgramName().equals(program.getId()) &&
+                scheduleProgramInfo.getProgramType().equals(program.getType().getSchedulableType());
+            }
+          });
+        replaceAppSpecInProgramJar(program, newAppSpec);
+        mds.apps.updateAppSpec(program.getNamespaceId(), program.getApplicationId(), newAppSpec);
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public void deleteSchedules(final Id.Namespace namespaceId) {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        for (ApplicationSpecification appSpec : getAllApplications(namespaceId)) {
+          ApplicationSpecification newAppSpec =
+            removeSchedulesFromApp(appSpec, new Predicate<ScheduleProgramInfo>() {
+              @Override
+              public boolean apply(@Nullable ScheduleProgramInfo scheduleProgramInfo) {
+                return true;
+              }
+            });
+//          replaceAppSpecInProgramJar(program, newAppSpec);
```

Comment:
Filed JIRA https://issues.cask.co/browse/CDAP-2957 to use app spec from MDS instead of the app spec in program jar. Once the JIRA is fixed, we'll no longer have to update program jar for changes to app spec.

In this case - if delete schedules for a namespace can only be called while deleting a namespace then there is no point in updating the program jars. I think it should be okay to add a comment for now, and not update the program jars.